app/data_model/database.py

```diff
@@ -20,19 +20,24 @@ class QuestionnaireState(base):
     __tablename__ = 'questionnaire_state'
     user_id = Column('userid', String, primary_key=True)
     state = Column('questionnaire_data', String)
+    data = None
 
     def __init__(self, user_id, data):
         self.user_id = user_id
+        self.data = data
         self.state = json.dumps(data)
```

Comment:
Now that we hold onto data I think we could populate state on access which will save a little bit of time as well. Probably justifies using an @property decorator.

Reply:
Im not sure its that easy is it when using sqlalchemy?